clients/mesh_rpc_client/src/ws_client.ts

```diff
@@ -0,0 +1,237 @@
+    }
+    /**
+     * Adds an array of 0x signed orders to the Mesh node.
+     * @param signedOrders signedOrders to add
+     * @returns validation results
+     */
+    public async addOrdersAsync(signedOrders: SignedOrder[]): Promise<ValidationResults> {
+        assert.isArray('signedOrders', signedOrders);
+        const rawValidationResults: RawValidationResults = await (this._wsProvider as any).send('mesh_addOrders', [
+            signedOrders,
+        ]);
+        const validationResults: ValidationResults = {
+            accepted: WSClient._convertRawAcceptedOrders(rawValidationResults.accepted),
+            rejected: [],
+        };
+        rawValidationResults.rejected.forEach(rawRejectedOrderInfo => {
+            const rejectedOrderInfo: RejectedOrderInfo = {
+                orderHash: rawRejectedOrderInfo.orderHash,
+                signedOrder: orderParsingUtils.convertOrderStringFieldsToBigNumber(rawRejectedOrderInfo.signedOrder),
+                kind: rawRejectedOrderInfo.kind,
+                status: rawRejectedOrderInfo.status,
+            };
+            validationResults.rejected.push(rejectedOrderInfo);
+        });
+        return validationResults;
+    }
+    /**
+     * Get all 0x signed orders currently stored in the Mesh node
+     * @param perPage number of signedOrders to fetch per paginated request
+     * @returns all orders, their hash and their fillableTakerAssetAmount
+     */
+    public async getOrdersAsync(perPage: number = 200): Promise<AcceptedOrderInfo[]> {
+        let snapshotID = ''; // New snapshot
+        let ordersInfosLen = 1;
+
+        let i = 0;
+        let rawAcceptedOrderInfos: RawAcceptedOrderInfo[] = [];
+        while (ordersInfosLen === perPage) {
+            const page = i;
+            const getOrdersResponse: GetOrdersResponse = await (this._wsProvider as any).send('mesh_getOrders', [
+                page,
+                perPage,
+                snapshotID,
+            ]);
+            snapshotID = getOrdersResponse.snapshotID;
+            rawAcceptedOrderInfos = [...rawAcceptedOrderInfos, ...getOrdersResponse.ordersInfos];
+            ordersInfosLen = getOrdersResponse.ordersInfos.length;
+            i++;
+        }
+        const allOrdersInfos = WSClient._convertRawAcceptedOrders(rawAcceptedOrderInfos);
+        return allOrdersInfos;
+    }
+    /**
+     * Subscribe to the 'orders' topic and receive order events from Mesh. This method returns a
+     * subscriptionId that can be used to `unsubscribe()` from this subscription.
+     * @param   cb   callback function where you'd like to get notified about order events
+     * @return subscriptionId
+     */
+    public async subscribeToOrdersAsync(cb: (orderEvents: OrderEvent[]) => void): Promise<string> {
+        assert.isFunction('cb', cb);
+        const orderEventsSubscriptionId = await this._wsProvider.subscribe('mesh_subscribe', 'orders', []);
+        const id = uuid();
```

Comment:
Upon re-connection, a new subscription is established, and it will have a new `subscriptionID` generated by Mesh. I create a new ID here that _will not_ change upon re-connection. This let's the caller have a consistent reference to the subscription they setup, regardless of re-connection. 

Yes, agreed that many callbacks could share the same subscription, but it's not obvious to me that this would be the expected behavior by the caller. Why not just set up a single one and then dispatch the events received from a single subscription to do multiple things?

Reply:
> Why not just set up a single one and then dispatch the events received from a single subscription to do multiple things?

Yes that's what I'm suggesting, though I'm suggesting the client does this rather than the user. Why make them do the extra effort?

Reply:
Agreed on the extra effort. Will refactor.

Reply:
It adds extra effort keeping track of the number of subscribers internally and it's not immediately obvious that it's a better design overall. Move clients will also only have a single subscription at a time anyways.